### PR TITLE
Add engine utils endpoint

### PR DESCRIPTION
**Pull request by mlfarrell**

### CODE REVIEW

File: modeling-cmds/src/def_enum.rs [hunk 1]
```diff
@@ -40,6 +40,18 @@ define_modeling_cmd_enum! {
...
 
+        /// Evaluates the position of a path in one shot (engine utility for kcl executor)
+        #[derive(
+            Debug, Clone, Serialize, Deserialize, JsonSchema, ModelingCmdVariant,
+        )]
+        pub struct EngineUtilEvaluatePath {
+            /// The path in json form (the serialized result of the kcl Sketch/Path object
+            pub path_json: String,
+
+            /// The evaluation parameter
```
Comment (adamchalmers) [inline modeling-cmds/src/def_enum.rs:51]: Could you give a bit more detail please? These docs explain how to use the endpoint, and right now I don't know how to use the endpoint.

File: modeling-cmds/src/ok_response.rs [hunk 2]
```diff
@@ -22,6 +22,13 @@ define_ok_modeling_cmd_response_enum! {
             units,
         };
 
+        /// The response of the `EngineUtilEvaluatePath` endpoint
+        #[derive(Debug, Serialize, Deserialize, Clone, JsonSchema, ModelingCmdOutput)]
+        pub struct EngineUtilEvaluatePath {
+            /// The evaluated position
```
Comment (adamchalmers) [inline modeling-cmds/src/ok_response.rs:28]: Which position along the path is this?

File: modeling-cmds/src/def_enum.rs (hunk 1 above)
Comment (mlfarrell) [reply modeling-cmds/src/def_enum.rs:51]: this is the exact "parameter" that i was talking about during my curves/nurbs talk at kitty camp.  its not a programming parameter, its a curve parameter.   I suppose we could say "path curve parameter?"

File: modeling-cmds/src/ok_response.rs (hunk 2 above)
Comment (mlfarrell) [reply modeling-cmds/src/ok_response.rs:28]: the evaluated one from the t value passed in.

**Pull request merged**